src/rest-server/src/models/v2/group.js

```diff
@@ -31,236 +30,182 @@ const crudConfig = crudGroup.initConfig(process.env.K8S_APISERVER_URI);
+  for (const userItem of userList) {
+    if (userItem['grouplist'].includes(groupname)) {
+      userItem['grouplist'].splice(userItem['grouplist'].indexOf(groupname), 1);
+      updateUserList.push(userItem);
     }
-    return ret;
-  } catch (error) {
-    throw error;
   }
+  if (updateUserList.length !== 0) {
+    logger.info('User list to be updated has been prepared.');
+    logger.info('Begin to update user\' group list.');
+    await Promise.all(updateUserList.map(async (userData) => {
+      await userModel.updateUser(userData['username'], userData);
+    }));
+    logger.info('Update group info successfully.');
+  } else {
+    logger.info('No user\' grouplist need to be updated.');
+  }
+  return ret;
 };
 
-const addGroupIntoAdminUser = async (groupname, groupValue) => {
-  try {
-    let userList = await userModel.getAllUser();
-    let updateUserList = [];
-    for (let userItem of userList) {
-      if (!userItem['grouplist'].includes(authConfig.groupConfig.adminGroup.groupname)) {
-        continue;
-      }
-      if (!userItem['grouplist'].includes(groupname)) {
-        userItem['grouplist'].push(groupname);
-        if (groupValue['extension']['groupType'] === 'vc' && !userItem['extension']['virtualCluster'].includes(groupname)) {
-          userItem['extension']['virtualCluster'].push(groupname);
-        }
-        updateUserList.push(userItem);
+const getListGroup = async (grouplist) => {
+  return Promise.all(grouplist.map(getGroup));
+};
+
+const batchUpdateGroups = async (groupItems) => {
+  return await Promise.all(groupItems.map(async (groupItem) => {
+    await updateGroup(groupItem.groupname, groupItem);
+  }));
+};
+
+const getVCsWithGroupinfo = async (groupItems) => {
+  let virtualClusters = new Set();
+  for (const groupItem of groupItems) {
+    if (groupItem.extension && groupItem.extension.acls) {
+      if (groupItem.extension.acls.admin) {
+        return Object.keys(await vcModel.prototype.getVcListAsyc());
+      } else if (groupItem.extension.acls.virtualClusters) {
+        virtualClusters = new Set([...virtualClusters, ...groupItem.extension.acls.virtualClusters]);
       }
     }
-    if (updateUserList.length !== 0) {
-      logger.info('User list to be updated has been prepared.');
-      logger.info('Begin to update user\' group list.');
-      await Promise.all(updateUserList.map(async (userData) => {
-        await userModel.updateUser(userData['username'], userData);
-      }));
-      logger.info('Update group info successfully.');
-    } else {
-      logger.info('No user\' grouplist need to be updated.');
-    }
-  } catch (error) {
-    throw error;
   }
+  return [...virtualClusters];
 };
 
-const createGroup = async (groupname, groupValue) => {
-  try {
-    const ret = await crudGroup.create(groupname, groupValue, crudConfig);
-    await addGroupIntoAdminUser(groupname, groupValue);
-    return ret;
-  } catch (error) {
-    throw error;
-  }
+const getGroupVCs = async (groupname) => {
```

Comment:
getGroupsVCs(['groupname']) === getGroupVCs('groupname') ?

Reply:
Yes, getGroupVCs is frequently used, so I provide a wrapper for it.